src/deepseek.rs

```diff
@@ -71,6 +72,81 @@ impl TryFrom<f32> for FrequencyPenalty {
     }
 }
 
+// Does this make sense or do we want to handle serialization somewhere else?
+#[derive(Debug)]
+pub enum Message {
+    System {
+        content: String,
+        name: Option<String>,
+    },
+    User {
+        content: String,
+        name: Option<String>,
+    },
+    Assistant {
+        content: Option<String>,
+        name: Option<String>,
+        prefix: Option<bool>,
```

Comment:
A prefix can only be specified for the last turn in the chat. Therefore, we should probably special case the last turn in Chat and make a separate enum with User and PrefixedAssistant.

Also I don't like how it would be possible to do a reasoning_content on a non-reasoner model - I'm fighting the urge to argue this library should initially just support deepseek-reasoner and make the scope focused on highly intelligent workflows